website/docs/guides/eks-getting-started.html.md

```diff
@@ -0,0 +1,594 @@
+self-contained and should not affect existing infrastructure, its always best
+to be cautious!
+
+~> **NOTE:** If you would rather see the full sample Terraform configuration
+for this guide rather than the individual pieces, it can be found at:
+https://github.com/terraform-providers/terraform-provider-aws/tree/master/examples/eks-getting-started
+
+### Cluster Name Variable
+
+The below sample Terraform configurations reference a variable called
+`cluster-name` (`var.cluster-name`) which is used for consistency. Feel free
+to substitute your own cluster name or create the variable configuration:
+
+```hcl
+variable "cluster-name" {
+  default = "terraform-eks-demo"
+  type    = "string"
+}
+```
+
+### Base VPC Networking
+
+EKS requires the usage of [Virtual Private Cloud](https://aws.amazon.com/vpc/) to
+provide the base for its networking configuration.
+
+~> **NOTE:** The usage of the specific `kubernetes.io/cluster/*` resource tags below are required for EKS and Kubernetes to discover and manage networking resources.
+
+The below will create a 10.0.0.0/16 VPC, two 10.0.X.0/24 subnets, an internet
+gateway, and setup the subnet routing to route external traffic through the
+internet gateway:
+
+```hcl
+resource "aws_vpc" "demo" {
+  cidr_block = "10.0.0.0/16"
+
+  tags = "${
+    map(
+     "Name", "terraform-eks-demo-node",
+     "kubernetes.io/cluster/${var.cluster-name}", "shared",
+    )
+  }"
+}
+
+resource "aws_subnet" "demo" {
+  count = 2
+
+  availability_zone = "${data.aws_availability_zones.available.names[count.index]}"
+  cidr_block        = "10.0.${count.index}.0/24"
+  vpc_id            = "${aws_vpc.demo.id}"
+
+  tags = "${
+    map(
+     "Name", "terraform-eks-demo-node",
+     "kubernetes.io/cluster/${var.cluster-name}", "shared",
+    )
+  }"
+}
+
+resource "aws_internet_gateway" "demo" {
+  vpc_id = "${aws_vpc.demo.id}"
+
+  tags {
+    Name = "terraform-eks-demo"
+  }
+}
+
+resource "aws_route_table" "demo" {
+  vpc_id = "${aws_vpc.demo.id}"
+
+  route {
+    cidr_block = "0.0.0.0/0"
+    gateway_id = "${aws_internet_gateway.demo.id}"
+  }
+}
+
+resource "aws_route_table_association" "demo" {
+  count = 2
+
+  subnet_id      = "${aws_subnet.demo.*.id[count.index]}"
+  route_table_id = "${aws_route_table.demo.id}"
+}
+```
+
+### Kubernetes Masters
+
+This is where the EKS service comes into play. It requires a few operator
+managed resources beforehand so that Kubernetes can properly manage other
+AWS services as well as allow inbound networking communication from your
+local workstation (if desired) and worker nodes.
+
+#### EKS Master Cluster IAM Role
+
+The below is an example IAM role and policy to allow the EKS service to
+manage or retrieve data from other AWS services. It is also possible to create
+these policies with the [`aws_iam_policy_document` data source](/docs/providers/aws/d/iam_policy_document.html)
+
+For the latest required policy, see the EKS User Guide.
```

Comment:
Nitpick, but link to the EKS User Guide may be useful here.

Reply:
Ha! Great catch. I was waiting for the official URL. I'll add it.